scripts/benchmarks/lib/bench.py

```diff
@@ -557,6 +563,95 @@ def run(self, path=None):
         ])
 
 
+class PersistentFrontierBenchmark(Benchmark):
+
+    name = MeasurementColumn("name", 0)
+    time = FieldColumn("time", 1, "[ms]")
+    category = TagColumn("category", 2)
+    branch = TagColumn("gitbranch", 3)
+
+    def name_header(self):
+        return self.name
+
+    def branch_header(self):
+        return self.branch
+
+    def __init__(self, frontier_zipped_db):
+        Benchmark.__init__(self, BenchmarkType.persistent_frontier)
+        self.frontier_zipped_db = frontier_zipped_db
+
+    def headers(self):
+        return [
+            PersistentFrontierBenchmark.name, PersistentFrontierBenchmark.time,
+            PersistentFrontierBenchmark.category,  PersistentFrontierBenchmark.branch
+        ]
+
+    def fields(self):
+        return [
+            PersistentFrontierBenchmark.time
+        ]
+
+    def parse(self, content, output_filename, influxdb, branch):
+        buf = io.StringIO(content)
+        lines = buf.readlines()
+        rows = []
+        category = "persistent-frontier"
+        rows.append(list(map(lambda x: x.name, self.headers())))
+        metadata_key = "avg_duration_ms"
+
+        # We are parsing output in logproc json format
+        # We look for metadata entry which holds the measurement
+        for line in lines:
```

Comment:
I suggest that we ensure that logs are either not written or are directed to `stderr`. Then I suggest to output result of execution as a single line of json with the values needed. This way the loop would be avoided, which presents a clearer solution in my opinion.

Reply:
Nice that will help a lot then!